### PR TITLE
Deberta sentiment analyzer

### PR DESCRIPTION
**Pull request by LeonardPuettmannKern**
PR checklist:
- [x] Tested by creator on localhost:8000/docs
- [x] Tested by creator on refinery
- [x] Tested by reviewer on localhost:8000/docs
- [x] Tested by reviewer on refinery
- [x] (If added) common code tested in notebook/ script
- [x] Conforms with the agreed upon code pattern

### CODE REVIEW

**Comment by LeonardPuettmannKern** [comment]
Implements #302

**Comment by JWittmeyer** [comment]
tested common code without success:

image

running it again didn't yield the result anymore. 
Maybe the request had some issues :D

File: classifiers/llm/deberta_review_classifier/code_snippet_common.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        for entry in json_response
+    ]
+    return list(result[0].keys())[0]
+
+# ↑ necessary bricks function 
+# -----------------------------------------------------------------------------------------
+# ↓ example implementation 
+
+def example_integration():
+    api = "<API_KEY_GOES_HERE>"
```
Comment (JWittmeyer) [inline classifiers/llm/deberta_review_classifier/code_snippet_common.md:19]: api_key

File: classifiers/llm/deberta_review_classifier/code_snippet_common.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+def deberta_review_classifier(text: str, api_key: str) -> dict:
+    headers = {"Authorization": f"Bearer {api_key}"}
+    response = requests.post("https://api-inference.huggingface.co/models/RashidNLP/Amazon-Deberta-Base-Sentiment", headers=headers, json={"inputs": text})
+    json_response = response.json()
+    result = [
+        {item["label"]: item["score"] for item in entry}
+        for entry in json_response
+    ]
+    return list(result[0].keys())[0]
```
Comment (JWittmeyer) [inline classifiers/llm/deberta_review_classifier/code_snippet_common.md:12]: not sure if understood correctly but this seems a bit weird for me.

i read it as:
convert response to a list of dicts with label & score
take the first entry of that list and then take the label.

No sorting which indicates that the json_response is provided sorted

Why don't we just take the first entry of the json response then?
e.g.` json_response[0][0]["label"]`

or if we are unsure regarding how deep it is actually nested maybe:
```
r = json_response
while not isinstance(r, dict):
    r = r[0]
if "label" not in r:
    r = "Unkown"
else:
    r = r["label"]
```

File: classifiers/llm/deberta_review_classifier/code_snippet_refinery.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+```python
+import requests
+import json
```
Comment (JWittmeyer) [inline classifiers/llm/deberta_review_classifier/code_snippet_refinery.md:3]: here json is imported, in the common snippet its not. Since we don't use the json library I think we don't need it

File: classifiers/llm/deberta_review_classifier/config.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        docker_image="none",
+        available_for=["refinery", "common"],
+        part_of_group=[
+            "llm",
+            "not_gdpr_compliant",
+        ],  # first entry should be parent directory
+        # bricks integrator information
+        integrator_inputs={
+            "name": "deberta_review_classifier",
+            "refineryDataType": RefineryDataType.TEXT.value,
```
Comment (JWittmeyer) [inline classifiers/llm/deberta_review_classifier/config.py:26]: i think we need an outputs part since it's a classifier
"outputs": [
                "Positive",
                "Neutral",
                "Negative",
            ]

File: classifiers/llm/deberta_review_classifier/config.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                "ATTRIBUTE": {
+                    "selectionType": SelectionType.CHOICE.value,
+                    "optional": "false",
+                    "addInfo": [
+                        BricksVariableType.ATTRIBUTE.value,
+                        BricksVariableType.GENERIC_STRING.value
+                    ]
+                },
+                "API_KEY": {
+                    "selectionType": SelectionType.CHOICE.value,
```
Comment (JWittmeyer) [inline classifiers/llm/deberta_review_classifier/config.py:37]: SelectionType.STRING.value,

File: classifiers/sentiment/vader_sentiment_classifier/code_snippet_refinery.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
 # expects labeling task to have labels ["positive" ,"neutral", "negative"]
 from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
 
-ATTRIBUTE: str = "text" # only text attributes
+ATTRIBUTE: str = "headline" # only text attributes
```
Comment (JWittmeyer) [inline classifiers/sentiment/vader_sentiment_classifier/code_snippet_refinery.md:5]: why was this changed to headline? we usually use text everywhere else

File: classifiers/llm/deberta_review_classifier/code_snippet_refinery.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+API_KEY: str = "<API_KEY_GOES_HERE>"
+ATTRIBUTE: str = "headline" # only text attributes
```
Comment (JWittmeyer) [inline classifiers/llm/deberta_review_classifier/code_snippet_refinery.md:6]: see above

File: classifiers/llm/deberta_review_classifier/code_snippet_refinery.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+def deberta_review_classifier(record):
+    """
+    Uses toxic-bert via Hugging Face Inference API to classify toxicity in text.
+    """
+    api_token = API_KEY
```
Comment (JWittmeyer) [inline classifiers/llm/deberta_review_classifier/code_snippet_refinery.md:12]: i dont think we need to reassign this variable to another variable and can use the API_KEY in the headers directly

File: classifiers/llm/deberta_review_classifier/__init__.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import requests
+import json
+from pydantic import BaseModel
+
+INPUT_EXAMPLE = {
+    "apiToken": "<API_KEY_GOES_HERE>",
+    "text": "This is a great product and I would by it again, 10/10 would recommend!"
```
[the remark this answers is not in the document]
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/__init__.py:7]: Fixed this!

File: classifiers/llm/deberta_review_classifier/code_snippet_common.md (hunk 1 above)
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/code_snippet_common.md]: Changed!

File: classifiers/llm/deberta_review_classifier/code_snippet_refinery.md (hunk 3 above)
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/code_snippet_refinery.md]: Removed. 👍

File: classifiers/llm/deberta_review_classifier/config.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        tabler_icon="MoodNerd",
+        min_refinery_version="1.7.0",
+        state=State.PUBLIC.value,
+        gdpr_compliant="false",
+        type="premium",
+        kern_token_proxy_usable="false",
+        docker_image="none",
+        available_for=["refinery", "common"],
+        part_of_group=[
+            "llm",
```
[the remark this answers is not in the document]
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/config.py:20]: Depends. DeBERTa is small compared to a massive GPT-4 for example, but it is still several hundred MB large and therefore currently too big to run via the endpoint service on DigitalOcean. So I would put it to the large language models. 👍

File: classifiers/llm/deberta_review_classifier/config.py [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+            "not_gdpr_compliant",
+        ],  # first entry should be parent directory
+        # bricks integrator information
+        integrator_inputs={
+            "name": "deberta_review_classifier",
+            "refineryDataType": RefineryDataType.TEXT.value,
+            "variables": {
+                "ATTRIBUTE": {
+                    "selectionType": SelectionType.CHOICE.value,
+                    "optional": "false",
```
[the remark this answers is not in the document]
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/config.py]: Removed!

File: classifiers/llm/deberta_review_classifier/config.py (hunk 4 above)
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/config.py:26]: Added!

File: classifiers/sentiment/vader_sentiment_classifier/code_snippet_refinery.md (hunk 6 above)
Comment (LeonardPuettmannKern) [reply classifiers/sentiment/vader_sentiment_classifier/code_snippet_refinery.md]: Yes this is correct. This shouldn't be here, anyway. Removed it!

File: classifiers/llm/deberta_review_classifier/code_snippet_refinery.md (hunk 7 above)
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/code_snippet_refinery.md]: Changed it!

File: classifiers/llm/deberta_review_classifier/code_snippet_refinery.md (hunk 8 above)
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/code_snippet_refinery.md]: Correct! Changed it.

File: classifiers/llm/deberta_review_classifier/config.py (hunk 5 above)
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/config.py]: Done.

File: classifiers/llm/deberta_review_classifier/config.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (JWittmeyer) [reply classifiers/llm/deberta_review_classifier/config.py:20]: works for me

File: classifiers/llm/deberta_review_classifier/code_snippet_common.md (hunk 2 above)
Comment (LeonardPuettmannKern) [reply classifiers/llm/deberta_review_classifier/code_snippet_common.md]: Implemented this. Thanks for the suggestion!

File: classifiers/llm/deberta_review_classifier/code_snippet_common.md
Comment (SvenjaKernAi) [inline classifiers/llm/deberta_review_classifier/code_snippet_common.md:1]: When I tried out the common-code, it gives mit only "Unknown" label:
"This is a great product and I would buy it again, 10/10 would recommend!" is Unkown
"Product broke immediately. Don't like it. :/" is Unkown

So, there might be a issue in the json_response

**Review by SvenjaKernAi — APPROVED** [review]
Nice one :)

**Comment by LeonardPuettmannKern** [comment]
Implements #302
